code/modules/hydroponics/compost_bin.dm

```diff
@@ -0,0 +1,192 @@
+#define BASE_BIOMASS_CAPACITY 1500
+#define BASE_COMPOST_CAPACITY 1500
+#define SOIL_COST 25
+#define SOIL /obj/item/stack/sheet/soil
+#define DECAY 0.2
+#define MIN_CONVERSION 10
+/**
+  * # compost bin
+  * used to make soil from plants.
+  * Doesn't have components.
+  */
+/obj/machinery/compost_bin
+	name = "compost bin"
+	desc = "A wooden bin for composting"
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "compost_bin-empty"
+	power_state = NO_POWER_USE
+	density = TRUE
+	anchored = TRUE
+	// amount of biomass in the compost bin
+	var/biomass = 0
+	// amount of compost in the compost bin
+	var/compost = 0
+	// The maximum amount of biomass the compost bin can store.
+	var/biomass_capacity = BASE_BIOMASS_CAPACITY
+	// The maximum amount of compost the compost bin can store.
+	var/compost_capacity = BASE_COMPOST_CAPACITY
+	// Is the bin currently composting
+	var/composting = FALSE
+
```

Comment:
Could you make all of these comments `//` into `///` so they get autodocced?
Also, you don't need to comment on all variables, use common sense and remove the comments on the variables that explain themselves well, like `composting`

Reply:
will do